### PR TITLE
add accessor methods to UserCredentials

### PR DESCRIPTION
**Pull request by pongad**
I'm not quite sure why getRefreshToken was made final. Made all final for consistency. Please let me know if you want to drop the modifier.

### CODE REVIEW

File: oauth2_http/java/com/google/auth/oauth2/UserCredentials.java [hunk 1]
```diff
@@ -129,7 +129,49 @@ public AccessToken refreshAccessToken() throws IOException {
...
+
+  /**
+   * Returns HTTP object used to get access tokens.
+   *
+   * @return HTTP transport
+   */
+  public final HttpTransport getHttpTransport() {
+    return transport;
+  }
+
```
Comment (anthmgoogle) [inline oauth2_http/java/com/google/auth/oauth2/UserCredentials.java]: This is for rarer extensibility and testing scenarios. I would not have this as a public property.

File: oauth2_http/java/com/google/auth/oauth2/UserCredentials.java [hunk 2]
```diff
@@ -129,7 +129,49 @@ public AccessToken refreshAccessToken() throws IOException {
...
+  /**
+   * Returns URI of the end point that provides tokens.
+   *
+   * @return end point URI
+   */
+  public final URI getUri() {
+    return tokenServerUri;
+  }
+
```
Comment (anthmgoogle) [inline oauth2_http/java/com/google/auth/oauth2/UserCredentials.java]: Same here, this is for rare extensiblity scenarios. If it is private or protected it should be a more specific name.

**Comment by anthmgoogle** [comment]
Thanks. Suggesting a couple of the more obscure properties be left off.

**Comment by pongad** [comment]
Thank you @anthmgoogle . Please take another look.

File: oauth2_http/java/com/google/auth/oauth2/UserCredentials.java [hunk 3]
```diff
@@ -129,7 +129,31 @@ public AccessToken refreshAccessToken() throws IOException {
...
+  /**
+   * Returns the refresh token resulting from a OAuth2 consent flow.
+   *
+   * @return refresh token
+   */
   public final String getRefreshToken() {
     return refreshToken;
   }
+
```
Comment (anthmgoogle) [inline oauth2_http/java/com/google/auth/oauth2/UserCredentials.java]: Unintentional whitespace?

**Comment by anthmgoogle** [comment]
LGTM.

File: oauth2_http/java/com/google/auth/oauth2/UserCredentials.java (hunk 3 above)
Comment (pongad) [inline oauth2_http/java/com/google/auth/oauth2/UserCredentials.java]: https://google.github.io/styleguide/javaguide.html#s4.6.1-vertical-whitespace seems to neither encourage nor discourage whitespace after the last member of the class.
To keep things the same, I'll remove it before merging.

**Pull request merged**